lib/src/adder.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// adder.dart
+// Implementation of Adder Module.
+//
+// 2023 June 1
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// An abstract class for all adder module.
+abstract class Adder extends Module {
+  /// The input to the adder pin [a].
+  Logic a;
```

Comment:
the `input` signals of the class should _not_ be publicly visible.  they should also probably be `final` or at least `late final`.  At least we should use `@protected` if we don't find a nice way to make them private.

same applies to the multiplier